[ktzy0305] iP

ktzy0305 opened the pull request:
# Skye 
> Skye's task management skills are so legendary that to-do lists everywhere have started bowing down in awe. 🏆🤖 #SkyeRules

With Skye, managing your tasks is a breeze. Heres why you should choose Skye:
- Skye is text-based, who needs fancy graphics anyway?
- Skye is as easy to use as ~~learning ABC~~ sending a message.
- Skye is so fast that it **bends the laws of space and time!** _(Okay, maybe not, but it's pretty quick)_

All you need to do is follow these simple steps:
1. Download Skye from [here](https://github.com/ktzy0305/ip/releases/tag/Level-9).
2. Double-click on the executable file.
3. Start adding tasks like a boss. 😎
4. Sit back and let Skye manage your tasks like a pro 😉

Here's a snippet of the `main` method!
```java
public static void main(String[] args) {
    new Skye("data/tasks.txt").run();
}
```

Features:
- [X] Managing tasks
- [X] Organizing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

ktzy0305 committed:
rename, greet, exit

ktzy0305 committed:
add echo functionality

ktzy0305 committed:
feat: add task and list tasks

ktzy0305 committed:
add task.java

ktzy0305 committed:
feat: add mark as done and unmark

ktzy0305 committed:
add classes Deadline.java, Event.java and ToDo.java

ktzy0305 committed:
feat: add todos, events and deadlines

ktzy0305 committed:
add test inputs and expected outputs

ktzy0305 committed:
Add DukeException.java

ktzy0305 committed:
Add error handling for incorrect inputs

ktzy0305 committed:
Add support for deleting tasks from the list

ktzy0305 committed:
add functionality to save tasks

ktzy0305 committed:
Merge branch 'branch-Level-7'

ktzy0305 committed:
add dates and times to Skye

ktzy0305 committed:
Merge branch 'branch-Level-8'

ktzy0305 committed:
remove data/tasks.txt

ktzy0305 committed:
add more OOP

ktzy0305 committed:
add packaging of classes

ktzy0305 committed:
Merge branch 'branch-Packages'

ktzy0305 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-JUnit

ktzy0305 committed:
adds gradle build scripts, JUnit tests and create jar file

ktzy0305 committed:
Merge branch 'branch-JUnit'

ktzy0305 committed:
adhere to Java coding standards.

ktzy0305 committed:
Add JavaDoc documentation for classes and methods

ktzy0305 committed:
add find task command

ktzy0305 committed:
Merge branch 'branch-Level-9'

ktzy0305 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/ui/UI.java

ktzy0305 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/commands/Command.java
#	src/main/java/duke/commands/DueCommand.java
#	src/main/java/duke/commands/HelpCommand.java
#	src/main/java/duke/commands/ListCommand.java
#	src/main/java/duke/data/TaskList.java
#	src/main/java/duke/data/exception/DukeException.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/ui/UI.java

ktzy0305 committed:
Merge branch 'master' into branch-JUnit

ktzy0305 committed:
add unit test cases for TaskDecoder.java

ktzy0305 committed:
Merge branch 'branch-JUnit'

ktzy0305 committed:
Merge branch 'master' into branch-Packages

ktzy0305 committed:
rename package name from duke to skye

ktzy0305 committed:
Merge branch 'branch-Packages'

ktzy0305 committed:
update readme title

ktzy0305 committed:
add checkstyle to the project and change import order

ktzy0305 committed:
Merge branch 'branch-A-CheckStyle'

geoffong11 commented on src/main/java/skye/data/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+     */
+    @Override
+    public String toString() {
+        return String.format("[%s] %s", getStatusIcon(), getDescription());
```
I like how the status icon and description are abstracted!

geoffong11 commented on src/main/java/skye/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+    public Command parse(String userInput) throws DukeException {
+        Matcher matcher = COMMAND_FORMAT.matcher(userInput.trim());
+        if (!matcher.matches()) {
+            throw new DukeException(DukeExceptionType.UNKNOWN_COMMAND);
```
I like how the exception is personalized!

geoffong11 commented on src/main/java/skye/commands/ByeCommand.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    /**
+     * When called, the program will exit.
```
Should put a verb in front

geoffong11 commented on src/main/java/skye/data/task/Deadline.java:
```diff
@@ -0,0 +1,83 @@
+    protected LocalDateTime by;
+
+    /**
+     * Initialises an deadline object.
```
Use American spelling i.e. Initializes

geoffong11 commented on src/main/java/skye/data/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+     * @param date The deadline of the task.
+     * @return A list of tasks due on the specified deadline.
+     */
+    public List<Task> showTasksDueOn(LocalDate date) {
```
I like how you use streams and go beyond what is necessary! Good job 👍

geoffong11 reviewed (commented):
Good job!

jx124 commented on src/main/java/skye/data/exception/DukeExceptionType.java:
```diff
@@ -0,0 +1,74 @@
+ */
+public enum DukeExceptionType {
+
+    DEADLINE_NO_DESCRIPTION(
+    "The description of a deadline cannot be empty.\n"
+            + "Correct usage: deadline <description> /by <date>"
+    ),
```
I believe wrapped lines should be indented by 8 spaces? 
```suggestion
    DEADLINE_NO_DESCRIPTION(
            "The description of a deadline cannot be empty.\n"
            + "Correct usage: deadline <description> /by <date>"
    ),
```

ktzy0305 replied:
Thank you for pointing this out!

jx124 commented on src/main/java/skye/guide/help.txt:
```diff
@@ -0,0 +1,26 @@
+Usage:
```
I like how well clear the guide is!

jx124 commented on src/main/java/skye/ui/UI.java:
```diff
@@ -0,0 +1,195 @@
+    public void showRemovedTask(Task task, List<Task> tasks) {
+        printMessage(
+            String.format(
+                "Noted. I've removed this task:\n %s\nNow you have %d tasks in the list",
+                task.toString(),
+                tasks.size()
```
I believe wrapped lines should be indented by 8 spaces?
```suggestion
                    "Noted. I've removed this task:\n %s\nNow you have %d tasks in the list",
                    task.toString(),
                    tasks.size()
```

jx124 commented on src/main/java/skye/parser/Parser.java:
```diff
@@ -0,0 +1,286 @@
+ */
+public class Parser {
+
+    public static final Pattern COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    public static final Pattern TASK_NUMBER_ARGS_FORMAT = Pattern.compile("(?<taskNumber>.+)");
+    public static final Pattern DEADLINE_ARGS_FORMAT =
+            Pattern.compile("(?<description>.*?)\\s+/by\\s+(?<deadline>.*)");
+    public static final Pattern EVENT_ARGS_FORMAT =
+            Pattern.compile("(?<description>.*?)\\s+/from\\s+(?<fromDate>.*?)\\s+/to\\s+(?<toDate>.*)");
```
I like how you used named capture groups!

jx124 reviewed (commented):
Great attempt! I like how the whole program is structured.

lynnlow175 commented on src/main/java/skye/parser/TaskDecoder.java:
```diff
@@ -0,0 +1,66 @@
+     * @return Task
+     * @throws DukeException When an invalid save format is encountered
+     */
+    public Task decode(String line) throws DukeException {
```
I like how the name of the method is very relevant and pretty much sums up the method function!

lynnlow175 replied:
I like how the name of the method is very relevant and pretty much sums up the method function!

lynnlow175 commented on src/main/java/skye/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Represents a generic command to handle adding of Task objects and its subclasses.
+ */
+public class AddTaskCommand extends Command {
```
Really like how you broke down the commands into the different specific command groups:)

lynnlow175 commented on src/main/java/skye/data/task/Task.java:
```diff
@@ -0,0 +1,98 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Happy to see a boolean-like name!

lynnlow175 commented on src/main/java/skye/data/task/Event.java:
```diff
@@ -0,0 +1,107 @@
+     * @param other The other event object we are checking for equivalence
+     * @return if the other event object is equivalent to the current object
+     */
+    @Override
```
good job on going the extra mile to check for duplicate tasks!

lynnlow175 reviewed (commented):
Overall, your code is really clean and modular. Keep up the great work!

ktzy0305 committed:
Add a GUI to the Skye chatbot

ktzy0305 committed:
Merge branch 'branch-Level-10'

ktzy0305 committed:
Improve code quality and fix coding standard violations

Improvements were made to the code quality of the existing code base.

Coding standard violations regarding indentation of wrapped lines and javadoc comments were identified.

A summary of what was done
* Ensure that wrapped lines are indented by 8 spaces.
* Ensure all naming complies with coding standards
* Removal of dead code

ktzy0305 committed:
Add assertion statements for decoder and parser

ktzy0305 committed:
Merge pull request #1 from ktzy0305/branch-A-Assertions

Add assertion statements for decoder and parser

ktzy0305 committed:
Merge pull request #2 from ktzy0305/branch-A-CodeQuality

Improve code quality and fix coding standard violations

ktzy0305 committed:
Add a yml file to define GitHub Actions workflow

Creates a workflow such that whenever there is a push or pull request
in the repository the following things will happen:

* An VM will start
* Check out the repository to the VM
* Validate gradle wrapper
* Set up JDK and JavaFX
* Build and check with gradle
* IO redirection

ktzy0305 committed:
Merge branch 'branch-A-CI'

ktzy0305 committed:
runtest.bat: Ammend entry point of application

Ammend entry point to reflect changes to the name of the application

Changes:
* Duke to Skye

ktzy0305 committed:
Add functionality to record and manage places

As part of the extension features, Skye can now record venues you have
in mind to book in the future.

Assumptions (MVP)
* Booking is for one entire day

Future Extensions
* Using generics for storage and list

ktzy0305 committed:
Merge pull request #3 from ktzy0305/branch-D-Places

Add functionality to record and manage places

ktzy0305 committed:
Add resizing of window and tweak alignment of dialog box

ktzy0305 committed:
add layer of abstraction for storage

ktzy0305 committed:
update command usage and error messages

ktzy0305 committed:
Merge branch 'branch-A-BetterGui'

ktzy0305 committed:
restore bye command feature

ktzy0305 committed:
Merge branch 'master' into branch-A-Personality

ktzy0305 committed:
change response message for some commands

ktzy0305 committed:
Merge branch 'branch-A-Personality'

ktzy0305 committed:
Add User Guide V0.1

ktzy0305 committed:
update user guide to v1.0

ktzy0305 committed:
Merge branch 'master' into branch-A-BetterGui

ktzy0305 committed:
Fix message truncation and formatting

ktzy0305 committed:
Merge branch 'branch-A-BetterGui'

ktzy0305 committed:
Ammend comments to comply to code quality

ktzy0305 committed:
Add test cases for VenueList.java and VenueDecoder.java

ktzy0305 committed:
Merge branch 'branch-A-MoreTesting'

ktzy0305 committed:
Fix a bug where help command returns a file path

ktzy0305 committed:
change Ui.png to a higher resolution image

ktzy0305 committed:
Create CNAME

ktzy0305 committed:
Delete CNAME
